### PR TITLE
Porting the `ASP.NET Core Empty` template from C# to VB

### PR DESCRIPTION
**Pull request by mterwoord**
This PR is not yet complete.

Pieces we know for sure are missing:
1. Optional placeholders in template
2. Overridable ASP.NET Core version. 
3. Many elements from template.json. We're not sure of all the options in the C# version of the template.json.
4. Integration in build script

This version is a first draft version. We hope to gather feedback in order to get VB version(s) of the templates working.

### CODE REVIEW

**Comment by jmarolf** [comment]
There has been significant restructuring to the `dev` branch, sorry! `Company.WebApplication1.vbproj` should be added to `templating/src/Microsoft.DotNet.Web.ProjectTemplates/` as `EmptyWeb-VisualBasic.vbproj.in` not `Microsoft.DotNet.Web.ProjectTemplates.2.0/content` as that folder doesn't exist anymore.

File: src/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/EmptyWeb-VB/Startup.vb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+		' This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+		Public Sub Configure(app As IApplicationBuilder, env As IHostingEnvironment)
+			If env.IsDevelopment() Then
+				app.UseDeveloperExceptionPage()
+			End If
+
+			app.Run(Async Function(context) 
+				              Await context.Response.WriteAsync("Hello World!")
+						  End Function)
```
Comment (jmarolf) [inline src/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/EmptyWeb-VB/Startup.vb]: Can we fixup the formatting here?

File: src/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/EmptyWeb-VB/.template.config/template.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+   "shortName": "web",
+   "thirdPartyNotices": "https://aka.ms/template-3pn",
+   "tags": {
+     "language": "VB",
+     "type": "project"
+   },
+   "sourceName": "Company.WebApplication1",
+   "preferNameDirectory": true,
+   "guids": [
+     "53bc9b9d-9d6a-45d4-8429-2a2761773502"
```
Comment (jmarolf) [inline src/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/EmptyWeb-VB/.template.config/template.json]: Are we sure this is the correct GUID for VB?

**Comment by jmarolf** [comment]
Looks like we're also missing several launch settings files that I believe are needed (see [here](https://github.com/aspnet/templating/tree/dev/src/Microsoft.DotNet.Web.ProjectTemplates/content/EmptyWeb-CSharp/.template.config))

**Comment by Eilon** [comment]
Hi, thank you for the PR, but at this time there are no plans to have built-in VB templates for ASP.NET Core, so I'm closing this PR.

**Comment by sjmulder** [comment]
What's the decision process on these sort of things? If the community builds it, will it not be included no matter what?

**Comment by Eilon** [comment]
@sjmulder the ASP.NET Core team discussed this and made a decision. We have very little expertise on VB, and whatever VB templates we have we need to keep up-to-date as we add features and change template contents. So, unfortunately, even if the community built it, we are unable to support it. However, the templating system in .NET Core and in VS is fully pluggable, so anyone can release VB templates (or any other templates) and have them show up side-by-side with all the other templates. We'd be happy to link to those templates from this repo (and probably from other places too).

**Comment by sjmulder** [comment]
Hi @Eilon, thanks for explaining. That makes sense. Keep up the good work on ASP.NET Core!

**Comment by obelink** [comment]
@AnthonyDGreen We didn't talk about where to put these templates. @mterwoord and I assumed that this repo was the best place to publish our first attempt. Please let us know when and where we can move our work. We will work on the feedback of @jmarolf and start with the next template. 
@klausLoeffelmann Can you or - shall we - move your WebAPI project also to the new repo? If we are more into the 'templating' thing, perhaps we can do it with less effort. What's your idea about it?

But to be honest. I think it's a very bad signal that @Eilon gives. Little expertise cannot never be an excuse to ignore a **Microsoft language** used by hundreds of thousands developers. If you don't have the expertise in the team, you need to get it. Just my 2 cts...

**Comment by KlausLoeffelmann** [comment]
@obelink Hey Andre,

I need some time to catch up and learn what you guys are talking about.
There are many things to do currently for us, and I need to prioritize. 

But on first glance I would agree, it is a bad signal which @Eilon gives. Said that, I understand - and I always said it in other contexts as well - that we **always** have to take limited resources for VB on MS side into account and come up with ways to compensate for that. So, without giving it any further thought, I'd like to repeat: Better have something feasible, but have SOMETHING, than only dream about a perfect VB world, and never have anything. Let's do it with less effort, then!

I'll hope to catch up soon!

**Comment by obelink** [comment]
Fully aggreed ;-)

**Pull request closed without merging**